### DISCUSSION TITLE
작업 중 스르르 나도 모르게 거북목이 된다면?! [TurtleNeck]

### DISCUSSION THREAD

**Discussion by KeenieY** [DeveloperAcademy-POSTECH/.github#54] [Show and tell]
## 작업 중 나도 모르게 거북목이 된다면 'TurtleNeck'을 사용해보세요.
안녕하세요! 3기 러너 키니입니다. 다들 작업할 때 자세는 괜찮으신가요...?
저희 팀은 작업하다보면 저희도 모르게 거북이가 되어있는데요...

이런 분들을 위해 이번 MC3 챌린지에서 거북목을 탐지해주는 앱 TurtleNeck을 만들었습니다.

TurtleNeck은 에어팟을 이용해 사용자의 자세가 흐트러질 때 알림을 보내주는 macOS 앱입니다.

![Slide 16_9 - 2](https://github.com/user-attachments/assets/be214e12-fdde-42fa-bde7-37072c7ea763)

**Comment by Junu-Park** [comment]
### 안녕하세요 3기 `아서`입니다
+) 귀여운 거북이 캐릭터 알림은 자세가 나빠진 이후에도 계속 나쁜 자세를 유지했을 때! 나타나요🐢
거북이는 바른 자세로 고치거나 클릭하게 되면 사라져요🐢
거북이는 다른 데스크탑 화면이나 다른 앱으로 옮겨도 사라지지 않아요🐢

**Comment by HappyDoran** [comment]
### 안녕하세요! 3기 러너 도란입니다! 

+) TurtleNeck은 기존의 앱과 달리 Popover로 동작을 하는 방식으로 구현되었습니다.🐢
사용자의 작업 환경을 직접적으로 가리지 않고 Menubar의 아이콘을 통해서 열고 닫을 수 있도록 NSPopover로 구현했고, 
PIP Window같은 경우에는 앱이 화면에 표시하는 창을 직접 구현하여 띄우는 NSWindow를 사용하여 구현했습니다🐢

**Comment by saejinpark** [comment]
안녕하세요! 3기 러너 ‘제패토’입니다!

+) TurtleNeck은 에어팟을 통하여 노트북 사용자의 자세를 개선해주는 앱앱니다! 이쁘게 봐주세요

**Comment by HyunJaeyeon** [comment]
#### 🧘🏻 저희 앱은 처음 실행 시 개개인 맞춤 바른자세를 등록하여 더 정확한 측정이 가능합니다!
<img width="380" alt="image" src="https://github.com/user-attachments/assets/d082a4af-16bd-450e-8d98-3b5f4cbabe8a">

```swift
// 자세 평균 구하는 함수
 private func calculateAveragePitch() -> Double {
        let total = pitchValues.reduce(0, +)
        let average = total / Double(pitchValues.count)
        return average
```
manager에서 다룬 pitch(각도)값을 받은 후 평균을 구해서 사용자 바른 자세값에 쏙 넣어줬어요:-)

훨씬 멋진 기능들이 많으니!!!!
저희 앱 나중에 사용해보러 오시요 ~~😁
